pkg/envoy/cds/builder.go

```diff
@@ -28,9 +31,139 @@ import (
 	"github.com/openservicemesh/osm/pkg/trafficpolicy"
 )
 
+type clusterBuilder struct {
+	proxyIdentity identity.ServiceIdentity
+
+	outboundMeshTrafficPolicy *trafficpolicy.OutboundMeshTrafficPolicy
+
+	inboundMeshTrafficPolicy *trafficpolicy.InboundMeshTrafficPolicy
+
+	sidecarSpec configv1alpha2.SidecarSpec
+
+	egressTrafficPolicy *trafficpolicy.EgressTrafficPolicy
+
+	envoyOutboundPassthroughCluster string
+
+	egressEnabled bool
+
+	metricsEnabled bool
+
+	tracingEnabled bool
```

Comment:
can we collapse enabled, and address into a single field, such that if address is not the empty string then it's enabled?

Reply:
that would make sense but we do have a default value set for tracing address and port even if the address is nil or observability isn't enabled: https://github.com/openservicemesh/osm/blob/c5967d6df3d61ae0f864324824bdfaa78c58cc78/pkg/utils/mesh_config.go#L46

Reply:
That makes sense on the meshconfig, but on the builder we can have it being an empty string still. I'll leave a comment on the code where it's set